### PR TITLE
test: add comprehensive ADO provider tests

### PR DESCRIPTION
**Pull request by mthalman**
## test: add comprehensive ADO provider tests

Adds thorough test coverage for the ADO provider extension (`packages/ado`):

- **`extension.test.ts`** — Tests extension activation lifecycle: core extension discovery, activation failures, API validation, provider registration, configuration change listeners, and deactivate.
- **`adoWorkItemProvider.extended.test.ts`** — Tests WIQL query construction, batch fetching (200-item batches), partial batch failure handling, malformed JSON responses, network errors, multi-project support, concurrency guard, edge cases for descriptions, periodic refresh clamping, background vs user-triggered auth, and detail URL construction.
- **`adoPrReviewProvider.extended.test.ts`** — Tests getUserId caching/error handling, PR fetch error handling, multi-project support, concurrency guard, URL encoding, periodic refresh edge cases, background vs user-triggered auth, and PR URL construction.

All 86 ADO tests pass. No production code changes.

### CODE REVIEW

File: packages/ado/src/test/extension.test.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    vi.mocked(workspace.getConfiguration).mockImplementation((section?: string) => {
+      if (section === 'workcenterAdo') {
+        return {
+          get: vi.fn((key: string, defaultValue?: any) => {
+            if (key === 'organization') return 'myorg';
+            if (key === 'projects') return ['ProjectA'];
+            if (key === 'refreshIntervalSeconds') return 300;
+            return defaultValue;
+          }),
+        } as any;
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/ado/src/test/extension.test.ts:52]: Fixed in dc51c78. Set \
efreshIntervalSeconds\ to \
